Can't compile tests locally

test environment

- os: wsl ubuntu24.04
- toolchain: riscv-gnu-toolchain release latest

command(same with ci)

- `cmake -S . -B build -GNinja -DCMAKE_BUILD_TYPE=RelWithDebInfo -DFIRST_PARTY_TESTS=TRUE`
- `ninja -C build all riscv_sim_rv32d_rvfi generated_smt_rv64f generated_docs_rv64d`

some log

```
[22/30] Compiling test_hello_world.c
FAILED: test/first_party/rv32d_test_hello_world.c.elf /home/trdthg/repo/sail/sail-riscv/build/test/first_party/rv32d_test_hello_world.c.elf 
cd /home/trdthg/repo/sail/sail-riscv/build/test/first_party && /home/trdthg/riscv/bin/riscv64-unknown-elf-gcc -march=rv32gc -mabi=ilp32d -mno-relax -ffreestanding -nostdlib -static -mcmodel=medany -g -Og -Wall -Werror -T /home/trdthg/repo/sail/sail-riscv/test/first_party/src/common/link.ld -o rv32d_test_hello_world.c.elf /home/trdthg/repo/sail/sail-riscv/test/first_party/src/test_hello_world.c /home/trdthg/repo/sail/sail-riscv/test/first_party/src/common/crt0.S /home/trdthg/repo/sail/sail-riscv/test/first_party/src/common/nanoprintf.c /home/trdthg/repo/sail/sail-riscv/test/first_party/src/common/runtime.c
/home/trdthg/riscv/bin/../lib/gcc/riscv64-unknown-elf/14.2.0/../../../../riscv64-unknown-elf/bin/ld: address 0xa0002010 of rv32d_test_hello_world.c.elf section `.stack' is not within region `if_ram'
/home/trdthg/riscv/bin/../lib/gcc/riscv64-unknown-elf/14.2.0/../../../../riscv64-unknown-elf/bin/ld: rv32d_test_hello_world.c.elf section `.text' will not fit in region `if_ram'
/home/trdthg/riscv/bin/../lib/gcc/riscv64-unknown-elf/14.2.0/../../../../riscv64-unknown-elf/bin/ld: address 0xa0002010 of rv32d_test_hello_world.c.elf section `.stack' is not within region `if_ram'
/home/trdthg/riscv/bin/../lib/gcc/riscv64-unknown-elf/14.2.0/../../../../riscv64-unknown-elf/bin/ld: region `if_ram' overflowed by 12544 bytes
/home/trdthg/riscv/bin/../lib/gcc/riscv64-unknown-elf/14.2.0/../../../../riscv64-unknown-elf/bin/ld: warning: rv32d_test_hello_world.c.elf has a LOAD segment with RWX permissions
collect2: error: ld returned 1 exit status
[24/30] Compiling test_minstret.S
FAILED: test/first_party/rv32d_test_minstret.S.elf /home/trdthg/repo/sail/sail-riscv/build/test/first_party/rv32d_test_minstret.S.elf 
cd /home/trdthg/repo/sail/sail-riscv/build/test/first_party && /home/trdthg/riscv/bin/riscv64-unknown-elf-gcc -march=rv32gc -mabi=ilp32d -mno-relax -ffreestanding -nostdlib -static -mcmodel=medany -g -Og -Wall -Werror -T /home/trdthg/repo/sail/sail-riscv/test/first_party/src/common/link.ld -o rv32d_test_minstret.S.elf /home/trdthg/repo/sail/sail-riscv/test/first_party/src/test_minstret.S /home/trdthg/repo/sail/sail-riscv/test/first_party/src/common/crt0.S /home/trdthg/repo/sail/sail-riscv/test/first_party/src/common/nanoprintf.c /home/trdthg/repo/sail/sail-riscv/test/first_party/src/common/runtime.c
/home/trdthg/riscv/bin/../lib/gcc/riscv64-unknown-elf/14.2.0/../../../../riscv64-unknown-elf/bin/ld: address 0xa0002010 of rv32d_test_minstret.S.elf section `.stack' is not within region `if_ram'
/home/trdthg/riscv/bin/../lib/gcc/riscv64-unknown-elf/14.2.0/../../../../riscv64-unknown-elf/bin/ld: rv32d_test_minstret.S.elf section `.text' will not fit in region `if_ram'
/home/trdthg/riscv/bin/../lib/gcc/riscv64-unknown-elf/14.2.0/../../../../riscv64-unknown-elf/bin/ld: address 0xa0002010 of rv32d_test_minstret.S.elf section `.stack' is not within region `if_ram'
/home/trdthg/riscv/bin/../lib/gcc/riscv64-unknown-elf/14.2.0/../../../../riscv64-unknown-elf/bin/ld: region `if_ram' overflowed by 12628 bytes
/home/trdthg/riscv/bin/../lib/gcc/riscv64-unknown-elf/14.2.0/../../../../riscv64-unknown-elf/bin/ld: warning: rv32d_test_minstret.S.elf has a LOAD segment with RWX permissions
collect2: error: ld returned 1 exit status
[27/30] Building C object c_emulator/CMakeFiles/riscv_sim_rv64d.dir/__/riscv_model_rv64d.c.o
ninja: build stopped: subcommand failed.
```

--- a/test/first_party/src/common/link.ld
+++ b/test/first_party/src/common/link.ld
@@ -13,9 +13,9 @@ MEMORY
 
 SECTIONS
 {
-  /* MMIO devices. PMA must make this uncacheable (doesn't
-     matter in Sail because there are no caches). */
-  .bss.mmio : { *(.bss.mmio) } >if_mmio
+  /* Go to the start of memory, otherwise GNU ld (but not lld) will
+  try to start allocating at 0. */
+  . = ORIGIN(if_ram);
 
   /* Stack first so in theory we get stack overlow detection. */
   .stack ALIGN(16) (NOLOAD) : {
@@ -47,4 +47,8 @@ SECTIONS
   /* Thread-local data */
   .tdata : { *(.tdata) } >if_ram
   .tbss : { *(.tbss) } >if_ram
+
+  /* MMIO devices. PMA must make this uncacheable (doesn't
+     matter in Sail because there are no caches). */
+  .bss.mmio : { *(.bss.mmio) } >if_mmio
 }

